presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveArrayFunctionQueries.java

```diff
@@ -45,4 +45,54 @@ public void testShuffle()
         this.assertQuerySucceeds("SELECT shuffle(array_sort(quantities)) FROM orders_ex");
         this.assertQuery("SELECT array_sort(shuffle(quantities)) FROM orders_ex");
     }
+
+    @Test
+    public void testArrayMatch()
+    {
+        final String errorMessage = ".*division by zero presto.default.divide.*";
+        final String expectTrue = "SELECT true";
+        final String expectFalse = "SELECT false";
+        final String expectNull = "SELECT null";
+        // test all_match
+        this.assertQuerySucceeds("SELECT all_match(quantities, x -> ((10 / x) > 2)) FROM orders_ex");
+        this.assertQueryFails("SELECT all_match(array[1, 2, 0], x -> ((10 / x) > 2))", errorMessage);
+        this.assertQuery("SELECT all_match(array[1, 5, 0], x -> ((10 / x) > 2))", expectFalse);
+        this.assertQuery("SELECT all_match(array[1, 2, null], x -> ((10 / x) > 2))", expectNull);
+        this.assertQuery("SELECT all_match(array[1, 2], x -> ((10 / x) > 2))", expectTrue);
+
+        this.assertQuery("SELECT all_match(array[array[1, 2], array[1]], x -> (cardinality(x) > 1))", expectFalse);
```

Comment:
This query is evaluated on the coordinator (via constant folding), hence, doesn't test any code in the worker. Let's use queries that do get executed on the worker.

Reply:
Fixed.